general/development/policies/codingstyle/frankenstyle.md

```diff
@@ -0,0 +1,173 @@
+---
+title: Frankenstyle
+tags:
+  - Plugins
+---
+'Frankenstyle component names' refers to the naming convention that is used to uniquely identify a Moodle plugin based on the type of plugin and its name.  They are used throughout the Moodle code (with a notable exception being the css class names in the themes).
+
+Martin Dougiamas invented the word 'frankenstyle' to describe this naming system which was invented by Petr Škoda.
+
+## Format
+
+Frankenstyle component names have a prefix and then a folder name, separated by an underscore.
+
+1. The prefix is determined by the type of plugin. For example, the prefix for an activity module is **mod**.
+1. The name is the folder name of the plugin, always lower case.  For example, the name for Quiz is **quiz**.
+
+So the frankenstyle component name for the quiz module is **mod_quiz**.
+
+## Plugin types
+
+See [Plugin types](/docs/apis/plugintypes/) page for the list of all supported plugin types in Moodle and their frankenstyle prefix.
+
+To get a definitive list in your version of Moodle 2.x, use a small Moodle script with <tt>print_object(get_plugin_types());</tt>.
+
+## Core subsystems
+
+Subsystems in Moodle are not plugins themselves but can be referred to using **core_xxx** where xxx is the subsystem name as defined in get_core_subsystems().
+
+You can see these being used in the @package parameter in phpdocs, or in the [webservice function names](https://docs.moodle.org/dev/Web_services_Roadmap). Core subsystems can provide own strings via a file stored in <tt>lang/en/{`subsystemname`}.php</tt>. Some of them have a dedicated location with libraries, autoloaded classes and other resources.
```

Comment:
There are few more places that now use the name since we wrote these docs and they're pretty imporant so I'd suggest adding them and turning it into a list:

```suggestion
Other places that you may see these being used include:

- the PHP Namespace used to autoload classes
- the prefix used in JavaScript module names
- the prefix used in template names
- the `@package` parameter in phpdocs
- the [webservice function names](https://docs.moodle.org/dev/Web_services_Roadmap)

Core subsystems can provide own strings via a file stored in `lang/en/{`subsystemname`}.php`. Some of them have a dedicated location with libraries, autoloaded classes and other resources.
```